src/shims/env.rs

```diff
@@ -0,0 +1,27 @@
+use rustc::ty::{layout::{Size, Align}, TyCtxt};
+use rustc_mir::interpret::Memory;
+
+use crate::*;
+
```

Comment:
Please define a type in this file that contains the `env_vars: HashMap<Vec<u8>, Pointer<Tag>>`. Currently, you have tight coupling between a field declared in `machine.rs` and function(s) defined here.

See `tls.rs` as an example, where `TlsData` is defined in that file and then embedded into the machine in `machine.rs`.

Reply:
Ideally, that field will be private, and this file will provide all the methods needed for the shims. I am hoping to move some complexity from `foreign_item.rs` to here.